source/common/common/utility.cc

```diff
@@ -79,6 +79,20 @@ bool StringUtil::atoul(const char* str, uint64_t& out, int base) {
   }
 }
 
+bool StringUtil::atol(const char* str, int64_t& out, int base) {
+  if (strlen(str) == 0) {
+    return false;
+  }
+
+  char* end_ptr;
```

Comment:
I think you want to set `errno = 0` here -- if strtol successfully parses a string that actually contains LONG_MIN or LONG_MAX you'll check for `errno == ERANGE` below and possibly return an incorrect error based on the previous value of errno.

e.g., if you run
```
int64_t v;
StringUtil::atol("9223372036854775808", v, 10);
StringUtil::atol("9223372036854775807", v, 10);
```
the second call will return false because errno is still set to ERANGE from the first call. (`atoul` has this bug as well.)